sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -251,8 +251,7 @@ object OptimizeSkewedJoin extends CustomShuffleReaderRule {
 
     case shj @ ShuffledHashJoinExec(_, _, joinType, _, _,
         ShuffleStage(left: ShuffleQueryStageExec),
-        ShuffleStage(right: ShuffleQueryStageExec), isSkewJoin)
-        if !isSkewJoin && supportedJoinTypes.contains(joinType) =>
```

Comment:
We don't need to check the join types now, as `tryOptimizeJoinChildren` will return None immediately if the join type is not supported.